.github/workflows/ci.yml

```diff
@@ -73,8 +77,32 @@ jobs:
       - name: Create a source distribution
         run: python setup.py sdist
       - name: Publish package to PyPI
-        if: startsWith(github.ref, 'refs/tags')
         uses: pypa/gh-action-pypi-publish@release/v1
         with:
           user: __token__
           password: ${{ secrets.PYPI_API_TOKEN }}
+
+  docker:
+    needs: [test]
+    runs-on: ubuntu-latest
+    if: startsWith(github.ref, 'refs/tags')
+    steps:
+    - name: Check out the repo
+      uses: actions/checkout@v2
```

Comment:
Where is this stored?  I'm not too familiar with GitHub actions as opposed to other CICD tools.  Honestly, this PR seems like a good place for me to learn that.

Reply:
@mattfazza what are you referring to? The CI is supposed to push to dockerhub.  This `actions/checkout` checks out this repository in a github VM and it will build the docker image and push it.

Reply:
I'm referring to the action.  Your docker stage uses an action called "checkout".  Where is that action defined?  That's the missing link for me.  I understand what this does, and I see the matrix in under the actions tab here on the repo, but the internals of the checkout@v2 are unknown to me.

Reply:
Ahh, Thanks for clarifying.  The action lives [here](https://github.com/marketplace/actions/checkout)